wandb/integration/langchain/wandb_tracer.py

```diff
@@ -14,197 +14,33 @@
-        include_stdout: bool = True,
-        additional_handlers: Optional[List["BaseCallbackHandler"]] = None,
-    ) -> None:
-        """Method provided for backwards compatibility. Please directly construct `WandbTracer` instead."""
-        message = """Global autologging is not currently supported for the LangChain integration.
-Please directly construct a `WandbTracer` and add it to the list of callbacks. For example:
-
-LLMChain(llm, callbacks=[WandbTracer()])
-# end of notebook / script:
-WandbTracer.finish()"""
-        wandb.termlog(message)
-
-    def __init__(self, run_args: Optional[WandbRunArgs] = None, **kwargs: Any) -> None:
-        """Initializes the WandbTracer.
-
-        Parameters:
-            run_args: (dict, optional) Arguments to pass to `wandb.init()`. If not provided, `wandb.init()` will be
-                called with no arguments. Please refer to the `wandb.init` for more details.
-
-        To use W&B to monitor all LangChain activity, add this tracer like any other langchain callback
-        ```
-        from wandb.integration.langchain import WandbTracer
-        LLMChain(llm, callbacks=[WandbTracer()])
-        # end of notebook / script:
-        WandbTracer.finish()
-        ```.
-        """
-        super().__init__(**kwargs)
-        self._run_args = run_args
-        self._ensure_run(should_print_url=(wandb.run is None))
-
-    @staticmethod
-    def finish() -> None:
-        """Waits for all asynchronous processes to finish and data to upload.
-
-        Proxy for `wandb.finish()`.
-        """
-        wandb.finish()
-
-    def _log_trace_from_run(self, run: "Run") -> None:
-        """Logs a LangChain Run to W*B as a W&B Trace."""
-        self._ensure_run()
-
-        root_span = safely_convert_lc_run_to_wb_span(run)
-        if root_span is None:
-            return
-
-        model_dict = None
-
-        # TODO: Uncomment this once we have a way to get the model from a run
-        # model = safely_get_span_producing_model(run)
-        # if model is not None:
-        #     model_dict = safely_convert_model_to_dict(model)
-
-        model_trace = trace_tree.WBTraceTree(
-            root_span=root_span,
-            model_dict=model_dict,
+from langchain.callbacks.tracers import WandbTracer  # noqa: E402, I001
+
+
+class WandbTracer(WandbTracer):
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        wandb.termwarn(
+            "This integration has been moved to langchain. The preferred method to enable the tracer is to set the "
+            "environment variable `LANGCHAIN_WANDB_TRACING=true`. Please refer to the langchain documentation at "
+            "https://python.langchain.com/en/latest/integrations/agent_with_wandb_tracing.html  In future, "
+            "please use: `from langchain.callbacks.tracers import WandbTracer` in place of this import. This import "
+            "will be removed in a v0.15.5 of wandb"
```

Comment:
Please rm the last sentence.

Reply:
@dmitryduev, I changed the message recently based on comments from @morganmcg1 , I have also removed the last sentence as per your suggestion.

Reply:
thanks. please also see my note above regarding using our deprecation framework.

Reply:
removed